File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String,Integer> set = new HashMap<>();
        for (String string : participant) 
        {
            set.put(string, set.getOrDefault(string, 0) + 1); 
        }

        for (String string : completion) 
        {
            set.put(string, set.get(string)-1);
        }
      
        for (String key : set.keySet()) 
        {
            if (set.get(key) == 1) 
            {
                answer = key;
                break; 
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    counts = {}
    for p in participant:
        counts[p] = counts.get(p, 0) + 1
    
    for c in completion:
        counts[c] -= 1 
    
    for k, v in counts.items():
        if v != 0:
            return k
        
    return ""